test/pipelining_commands_test.rb

```diff
@@ -199,6 +199,17 @@ def test_hgetall_in_a_pipeline_returns_hash
     assert_equal result.first, { "field" => "value" }
   end
 
+  def test_zpopmax_in_a_pipeline_produces_future
+    r.zadd("sortedset", 1.0, "value")
+    future = nil
+    result = r.pipelined do
+      future = r.zpopmax("sortedset")
+    end
+
+    assert_equal [["value", 1.0]], result
+    assert_equal ["value", 1.0], future.value
```

Comment:
This test needs to be wrapped in `target_version('5.0.0') do`, so it's skipped on older redis versions.